frontend/gateway/pb/caps.go

```diff
@@ -0,0 +1,48 @@
+package moby_buildkit_v1_frontend
+
+import "github.com/moby/buildkit/util/apicaps"
+
+var Caps apicaps.CapList
+
+// Every backwards or forwards non-compatible change needs to add a new capability row.
+// By default new capabilities should be experimental. After merge a capability is
+// considered immutable. After a capability is marked stable it should not be disabled.
+
+const (
+	CapSolveBase         apicaps.CapID = "solve.base"
+	CapSolveInlineReturn apicaps.CapID = "solve.inlinereturn"
+	CapResolveImage      apicaps.CapID = "resolveimage"
+	CapReadFile          apicaps.CapID = "readfile"
+)
+
+func init() {
+
+	Caps.Init(apicaps.Cap{
+		ID:      CapSolveBase,
+		Enabled: true,
+		Status:  apicaps.CapStatusExperimental,
+	})
+
+	Caps.Init(apicaps.Cap{
+		ID:         CapSolveInlineReturn,
+		Name:       "inline return from solve",
+		Enabled:    true,
+		Deprecated: true,
+		Status:     apicaps.CapStatusExperimental,
```

Comment:
What does "Experimental" mean for deprecated cap?
Why not `Status: apicaps.CapStatusDeprecated`?

Reply:
These are just for tracking the status of a cap for humans and not really needed for code logic (except when showing the error message). I guess usually they would change at the same time but I don't think we need to limit it. Eg. when an experimental featurev1 is replaced by experimental featurev2, if there is no extra maintenance cost on keeping featurev1 we could deprecate and keep it around for a couple of months to allow people time to upgrade.